src/clang-c-frontend/clang_c_convert.h

```diff
@@ -231,6 +232,29 @@ class clang_c_convertert
    *   field: clang AST representing the class/struct/union field we are dealing with
    */
   bool is_field_global_storage(const clang::FieldDecl *field);
+
+  /*
+   * check if a method is constructor or destructor
+   * Arguments:
+   *  fd: clang AST representing a C++ method
+   */
+  bool is_ConstructorOrDestructor(const clang::FunctionDecl &fd);
+
+  /*
+   * Check whether a member function call refers to
+   * a virtual/overriding method.
+   * For C, it always return false.
+   * For C++, see the overriding method in clang_cpp_convert module
+   */
+  virtual bool check_member_expr_virtual_overriding(const clang::Decl &decl);
```

Comment:
The "check" and "expr" parts in the name of this method are not clear to me. Could we rename this function, maybe something like `is_virtual_method_overridden` or similar? Not clear whether that's the intent, though.

It would also be good if the comment explained what this method is supposed to do here, and not in some derived class - the interface is defined here. (This remark also applies to the comment on `get_vft_binding_expr` below.)

Reply:
Good point. 
Done. Renamed to `perform_virtual_dispatch`... hope this name is more meaningful and intuitive.